src/components/Forms/PhoneNumberField.js

```diff
@@ -0,0 +1,300 @@
+  },
+  halfWidth: {
+    width: '162px'
+  }
+}
+
+const getSnackStyles = snacksTheme => {
+  const { action } = snacksTheme.colors
+  return {
+    highlight: {
+      border: `1px solid ${action}`
+    }
+  }
+}
+
+const getInputSyles = ({props, theme, isFocused}) => {
+  const snacksStyles = getSnackStyles(theme)
+  const { disabled, hasError, inputStyle } = props
+  const disabledStlyes = disabled ? styles.inputDisabled : {}
+  const errorStyles = !disabled && hasError ? styles.inputError : {}
+  const focusedStyles = isFocused && !hasError ? snacksStyles.highlight : {}
+
+  return {
+    ...styles.input,
+    ...inputStyle,
+    ...disabledStlyes,
+    ...errorStyles,
+    ...focusedStyles
+  }
+}
+
+@withTheme
+@FormComponent
+@Radium
+class PhoneNumberField extends React.Component {
+  static propTypes = {
+    /** Name of the field */
+    name               : PropTypes.string.isRequired,
+    /** HTML autocomplete attribute */
+    autoComplete       : PropTypes.string,
+    /** DefaultValue for non controlled component */
+    defaultValue       : PropTypes.any,
+    /** Disable the text field */
+    disabled           : PropTypes.bool,
+    /** Text of label that will animate when TextField is focused */
+    floatingLabelText  : PropTypes.string,
+    /** Sets width to 100% */
+    fullWidth          : PropTypes.bool,
+    /** Sets width to 162px */
+    halfWidth          : PropTypes.bool,
+    /** FormComponent error for validation */
+    hasError           : PropTypes.bool,
+    /** Helper text will show up in bottom right corner below TextField */
+    helperText         : PropTypes.string,
+    /** Uniq id for input */
+    id                 : PropTypes.string,
+    /** Style for input */
+    inputStyle         : PropTypes.object,
+    /** Set by FormComponent by default.   */
+    isValid            : PropTypes.bool,
+    /** onFocus callback */
+    onFocus            : PropTypes.func,
+    /** onChange callback */
+    onChange           : PropTypes.func,
+    /** onBlur callback */
+    onBlur             : PropTypes.func,
+    /** onKeyDown callback */
+    onKeyDown          : PropTypes.func,
+    /** Mark the field as required.  */
+    required           : PropTypes.bool,
+    /** Error from server to show ServerError message */
+    serverError        : PropTypes.string,
+    /** Wrapper styles */
+    style              : PropTypes.object,
+    /** Text to show for validation error  */
+    validationErrorText: PropTypes.string,
+    /** Value will make TextField a controlled component  */
+    value              : PropTypes.string,
+    /** Snacks theme attributes provided by `Themer` */
+    snacksTheme        : themePropTypes
+  }
+
+  static defaultProps = {
+    autoComplete: 'on',
+    disabled    : false,
+    defaultValue: null,
+    onKeyDown: () => {} // eslint-disable-line no-empty-function
+  }
+
+  state = {
+    hasValue: this.props.defaultValue !== null || Boolean(this.props.value)
```

Comment:
Is it worth it to look into trying to consolidate some of this with the existing text input? They do a few similar things (like this default state, similar blur/focus/input change logic, etc.). Maybe not as part of this PR but in general we can look into extracting some core piece we could use in both.

Reply:
Yeah, I agree. I imagine there will be at least one or two other inputs we'll add that could share this logic as well